### PR TITLE
Chore/recover linters

### PR DESCRIPTION
**Pull request by dvilelaf**
## Proposed changes

* Fixes some linters
* Recovers the api docs check
* Runs api doc generator (therefore changes +120 files)

## Fixes

n/a

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](https://github.com/fetchai/agents-aea/blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -193,6 +193,8 @@ jobs:
         run: sudo npm install -g markdown-spellcheck
       - name: Check Docs links and IPFS hashes
         run: tox -e check-doc-links-hashes
+      - name: Check API Docs updated
+        run: tox -e check-api-docs
```
Comment (dvilelaf) [inline .github/workflows/workflow.yml:197]: Recovering the check-api-docs

File: .pylintrc [hunk 2]
```diff
@@ -21,6 +21,19 @@ disable=C0103,C0201,C0301,C0302,W0105,W0707,W1202,W1203,R0801,E1136,E0611,C0209,
...
+# R1735: use-dict-literal, # kept as no harm
+# R1732: consider-using-with
+# W0237: arguments-renamed
+# R1733: unnecessary-dict-index-lookup
+# R1710: inconsistent-return-statements
+# R1714: consider-using-in
+# R1729: use-a-generator
+# R0402: consider-using-from-import
+# W0640: cell-var-from-loop
+# C0206: consider-using-dict-items
```
Comment (dvilelaf) [inline .pylintrc:36]: All these were introduced between pylint 2.6.0 to 2.11.1. Now we have introduced Tomte and making that upgrade, we do not want to change our linter settings so these must be skipped.

File: aea/helpers/async_utils.py [hunk 3]
```diff
@@ -412,7 +412,7 @@ def start(self) -> bool:
             self._thread = Thread(
                 target=self._thread_target, name=self.__class__.__name__  # type: ignore # loop was set in set_loop
             )
-            self._thread.setDaemon(True)
+            self._thread.setDaemon(True)  # pylint: disable=deprecated-method
```
Comment (dvilelaf) [inline aea/helpers/async_utils.py:415]: There was only 1 of these errors, so I skipped it locally.

Comment (Karrenbelt) [reply aea/helpers/async_utils.py:415]: https://github.com/python/cpython/pull/25174

`threading.Thread.setDaemon` => :attr:`threading.Thread.daemon`

File: aea/helpers/async_utils.py [hunk 4]
```diff
@@ -412,7 +412,7 @@ def start(self) -> bool:
             self._thread = Thread(
                 target=self._thread_target, name=self.__class__.__name__  # type: ignore # loop was set in set_loop
             )
-            self._thread.setDaemon(True)
+            self._thread.daemon = True
```
Comment (angrybayblade) [inline aea/helpers/async_utils.py:415]: Why was this changed?

File: tests/test_cli/test_registry/test_utils.py [hunk 5]
```diff
@@ -277,7 +277,7 @@ def test_extract_wrong_file_type(self):
         """Test for extract method wrong file type."""
         source = "file.wrong"
         target = "target-folder"
-        with self.assertRaises(Exception):
+        with self.assertRaises(ValueError):
```
Comment (angrybayblade) [inline tests/test_cli/test_registry/test_utils.py:280]: Same question

Comment (dvilelaf) [reply tests/test_cli/test_registry/test_utils.py:280]: Pylint informed that this check was too generic. I could skip the failure, but it seemed very easy to solve. Would you prefer to revert this change and skip the lint error?

Comment (angrybayblade) [reply tests/test_cli/test_registry/test_utils.py:280]: If the test pass no need to revert the change

File: aea/helpers/async_utils.py (hunk 4 above)
Comment (DavidMinarsch) [inline aea/helpers/async_utils.py:415]: Was this an api change?

**Review by DavidMinarsch — APPROVED** [review]

File: aea/helpers/async_utils.py (hunk 4 above)
Comment (dvilelaf) [reply aea/helpers/async_utils.py:415]: A deprecation: https://github.com/valory-xyz/open-aea/pull/371#discussion_r995461494
